docs-src/guides/add-new-blockchain.md

```diff
@@ -0,0 +1,66 @@
+- linking.
+
+*Authentication.* IdentityWallet (and 3id-connect) create `3id` (Ceramic flavour of DID) private keys
+based on an externally-provided entropy. It could be provided by a blockchain account by merely
+signing a well-known message. From a user's standpoint,
+it is authentication _into_ Ceramic through her blockchain account, be it on Ethereum, Filecoin,
+EOS, Cosmos or something else. Same signature (=same entropy) generates same Ceramic DID.
+
+*Linking.* In addition to generating DID a user could also _link_ additional blockchain accounts to Ceramic DID.
+It establishes a relation `blockchain account → DID` that allows one to discover DID along with social profile
+based on just a blockchain account. Additionally, a link serves as a proof-of-ownership by DID over the blockchain account.
+This has proven to be useful fir dApp personalization: one sees familiar names instead of `0xgibberish`.
+
+Below one additional process is mentioned: validation. It checks if _proof-of-ownership_ in the link is formally correct,
+i.e., a well-known payload is really signed by the account that is declared in the link.
+
+## Adding a new blockchain
+
+To add a new blockchain one have to do both linking and validation sections. 
+
+### Linking
+
+We use [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) to represent accounts in a blockchain agnostic way.
+If the blockchain you want to add isn't already part of the CAIP standards you should make sure to add it there.
+
+To add a new blockchain, one has to implement a new class implementing [AuthProvider](https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/blockchain-utils-linking/src/auth-provider.ts), put it
+into [`@ceramicnetwork/blockchain-utils-linking`](https://github.com/ceramicnetwork/js-ceramic/tree/develop/packages/blockchain-utils-linking) package and export it.
+See existing auth providers for inspiration.
+
+Auth provider is expected to be mainly called in a web browser as part of [3id-connect](https://github.com/3box/3id-connect) flow.
+Auth provider sits between 3id-connect (or IdentityWallet) and your blockchain account provider. In case of Ethereum,
+it might be MetaMask. Auth provider is responsible mainly for:
+- authentication (`#authenticate`): provide entropy,
+- creating link (`#createLink`): create a LinkProof object which associates the specified AccountID with the DID.
+
+Auth provider is expected to know which blockchain account it currently serves. It reports it via `#accountId`.
+To reuse same internal settings, e.g. connection to blockchain provider, but with a different account,
+auth provider should have `#withAddress` method.
+
+To sum it all up, to add a new blockchain for linking:
+- add new class implementing `AuthProvider` interface to `@ceramicnetwork/blockchain-utils-linking` package,
+- its `authenticate` method should ask blockchain provider for an authentication signature and return `0x`-prefixed string,
```

Comment:
I think this should be a `0x`-prefixed hex string right? Should we also say that it should be 32 bytes?

Reply:
AFAIK, this goes through sha256 to really create an entropy, so it could be of any length.

https://github.com/3box/3id-connect/blob/typescript/src/connectService.ts#L201

Reply:
It is good though to still make that requirement in the docs to avoid problems down the line.

Reply:
"...and return 32 bytes in form of `0x`-prefixed hex string"

Reply:
Thanks.